k2/csrc/util.h

```diff
@@ -7,15 +7,25 @@
 #ifndef K2_CSRC_UTIL_H_
 #define K2_CSRC_UTIL_H_
 
+#include <algorithm>
 #include <cfloat>
 #include <cmath>
 #include <functional>
+#include <limits>
 #include <utility>
 
 #include "k2/csrc/fsa.h"
 
 namespace k2 {
 
+#define EXPECT_DOUBLE_ARRAY_APPROX_EQ(expected, actual, abs_error)             \
+  EXPECT_EQ(expected.size(), actual.size()) << "Different Array Size.";        \
+  for (std::size_t i = 0; i < std::min(expected.size(), actual.size()); ++i) { \
```

Comment:
Why is there an `std::min`?
`EXPECT_EQ` has already been used to check that they have the same size.

BTW: `EXPECT_EQ` is from `gtest`, so this may not be the best place to place
the macro `EXPECT_DOUBLE_ARRAY_APPROX_EQ`.

Reply:
as `EXPECT_EQ` will not terminate the program, maybe I should use `ASSERT_EQ`. 

`EXPECT_TRUE` is also from gtest, seems we need to write our own check for this?

Reply:
I find that this macro is used only in `weights_test.cc`, should we place it in that file?

Reply:
It will be definitely used in other algortihms like `Rm/Intersect-Prue` in `fsa_algo.cc`.

I'm open with it, what do you think? write our own CHECK for this?